### PR TITLE
[exporter/prometheusremotewrite] support disabling target_info metric

### PR DESCRIPTION
**Pull request by hardproblems**
**Description:** <Describe what has changed.>
Add the ability to turn off the target_info metric to address duplicate series / out of order samples issue below

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12300

**Testing:** <Describe what testing was performed and which tests were added.>
- added unit test in pkg/translator/prometheusremotewrite/helper.go

**Documentation:** <Describe the documentation added.>
- added a comment in config.go
- added explanation for the param in README

@dashpole @Aneurysm9 @bogdandrutu @gouthamve @jmacd @jsuereth

### CODE REVIEW

File: exporter/prometheusremotewriteexporter/config.go
Comment (dashpole) [inline exporter/prometheusremotewriteexporter/config.go:53]: This name (disable_target_info_metric) differs from the name in the readme.

Comment (hardproblems) [reply exporter/prometheusremotewriteexporter/config.go:53]: oops fixed!

**Review by dashpole — APPROVED** [review]

**Comment by jmacd** [comment]
Speculating that `target_info` has a bug. 

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12300#issuecomment-1196934652

I would probably add to the current specification that if the target_info has no fields, it should be automatically omitted.

**Comment by jmacd** [comment]
Shouldn't you have the option remove this at the source? You've described a single-writer violation happening and I'd rather it be removed before it enters the OTLP pipeline. Otherwise, receivers other than PRW will face the same single-writer violations.

**Comment by dashpole** [comment]
It is possible today to  drop`target_info` in the prometheus receiver using `metric_relabel_configs`:

```yaml
metric_relabel_configs:
- source_labels: [__name__]
  regex: target_info
  action: drop
```

There is an added issue that we add "implicit" resource attributes (e.g. `net.host.name`) from `instance` (described in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9967) which produces an outgoing `target_info` metric even if you dropped the incoming `target_info` in the prometheus receiver.  It is also common for users to use automatic resource detectors (e.g. the `gcp` detector) to add resource information.  This would end up appearing in the `target_info` metric (which is useful!) and may cause the `target_info` collision if `job` and `instance` are not identifying.

**Comment by hardproblems** [comment]
>Shouldn't you have the option remove this at the source, i.e., in the prometheus receiver? You've described a single-writer violation happening and I'd rather it be removed before it enters the OTLP pipeline. Otherwise, receivers other than PRW will face the same single-writer violations.

We're using the otlp receiver. I can try using resource attributes processor to drop otel sdk added resources. iirc job/instance labels are likely added by the exporter so no way to drop those using processors.

**Comment by jmacd** [comment]
> job/instance labels are likely added by the exporter so no way to drop those using processors.

This sounds like a problem worth fixing. When `process_info` has only two attributes "job" and "instance" it is redundant with `up` as generated by the prometheus receiver. I would support a PRWexporter option to not generate `process_info`, therefore, when it contains only `job` and `instance`.

**Comment by dashpole** [comment]
> This sounds like a problem worth fixing. When process_info has only two attributes "job" and "instance" it is redundant with up as generated by the prometheus receiver. I would support a PRWexporter option to not generate process_info, therefore, when it contains only job and instance.

Do you mean `target_info` instead of `process_info`?  If so, I agree.  That was the initial intention, but it doesn't look like it was implemented that way.  We should be [checking if there are no attributes](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/aa86130be3728efb3ee68869082bcf7409fa46f1/pkg/translator/prometheusremotewrite/helper.go#L493) _after_ the RemoveIf that filters out job, instance.

**Comment by jmacd** [comment]
@dashpole yes, I've corrected the comment. Thanks.

File: exporter/prometheusremotewriteexporter/config.go
Comment (mx-psi) [inline exporter/prometheusremotewriteexporter/config.go:53]: Is it possible that we would add further customization options for `target_info`? If so I would rework this as:

```
target_info:
  enabled: <true|false>
```

cc @dashpole , not sure if there are specific conventions about this on this exporter

Comment (hardproblems) [reply exporter/prometheusremotewriteexporter/config.go:53]: I can update this config. Questions: should `enabled` default to true to keep with the existing release or are you ok with defaulting this to false (would it be considered breaking to default to false)?

Comment (hardproblems) [reply exporter/prometheusremotewriteexporter/config.go:53]: also would you want to update the way it's passed into prwExporter or are you ok to keep that the same?

Comment (hardproblems) [reply exporter/prometheusremotewriteexporter/config.go]: @max-psi updated config. lmk whether you want to prwExporter to also be changed 🙇

Comment (dashpole) [reply exporter/prometheusremotewriteexporter/config.go]: It should be defaulted to true.

**Review by dashpole — APPROVED** [review]

**Comment by hardproblems** [comment]
@mx-psi updated target_info config to use a struct. PTAL 🙇

**Comment by hardproblems** [comment]
@Aneurysm9 PTAL as well 🙇

**Review by Aneurysm9 — APPROVED** [review]

**Comment by dashpole** [comment]
```
config_test.go:98:16: cannot use (TargetInfo literal) (value of type TargetInfo) as *TargetInfo value in struct literal (typecheck)
			TargetInfo: TargetInfo{
```

**Comment by hardproblems** [comment]
@dashpole  @mx-psi @Aneurysm9 can I get some help merging this? I'm not authorized to merge and I have to keep merging main to this while waiting for a stamp to trigger CI checks.

**Comment by bogdandrutu** [comment]
@dashpole there are lint errors in this PR.

**Comment by dashpole** [comment]
@bogdandrutu agh, sorry.  I thought the author had fixed it, but I had to approve the tests.

**Comment by dashpole** [comment]
@hardproblems you can hold off rebasing on main for a bit (unless there is a conflict).  The maintainers can do that before they merge.  Its hard to tell if your tests are passing or not...

**Comment by hardproblems** [comment]
@dashpole @Aneurysm9 @mx-psi @bogdandrutu what's the process for getting this merged? I've looked through the docs https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md and https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/release.md and am still unclear what the process is after the code reviews are done and the CI checks appear to be passing.

**Pull request merged**
